application/src/main/java/org/togetherjava/tjbot/commands/moderation/modmail/ModMailCommand.java

```diff
@@ -0,0 +1,184 @@
+                CommandVisibility.GLOBAL);
+
+        OptionData messageOption = new OptionData(OptionType.STRING, OPTION_MESSAGE,
+                "What do you want to tell them?", true);
+        OptionData guildOption = new OptionData(OptionType.STRING, OPTION_GUILD,
+                "The server to contact mods from", true);
+        OptionData anonymousOption = new OptionData(OptionType.BOOLEAN, OPTION_STAY_ANONYMOUS,
+                "If set, your name is hidden - note that mods then can not get " + "back to you",
+                true);
+
+        List<Command.Choice> choices = jda.getGuildCache()
+            .stream()
+            .map(guild -> new Command.Choice(guild.getName(), guild.getIdLong()))
+            .toList();
+
+        guildOption.addChoices(choices);
+
+        getData().addOptions(messageOption, guildOption, anonymousOption);
+
+        modMailChannelNamePredicate =
+                Pattern.compile(config.getModMailChannelPattern()).asMatchPredicate();
+    }
+
+    private Cache<Long, Instant> createCooldownCache() {
+        return Caffeine.newBuilder()
+            .maximumSize(1_000)
+            .expireAfterAccess(COOLDOWN_DURATION_VALUE, TimeUnit.of(COOLDOWN_DURATION_UNIT))
+            .build();
+    }
+
+    @Override
+    public void onSlashCommand(SlashCommandInteractionEvent event) {
+        long userId = event.getUser().getIdLong();
+
+        if (handleIsOnCooldown(userId, event)) {
+            return;
+        }
+        authorToLastModMailInvocation.put(userId, Instant.now());
+        event.deferReply().setEphemeral(true).queue();
+
+        long userGuildId = event.getOption(OPTION_GUILD).getAsLong();
+        Optional<TextChannel> modMailAuditLog = getModMailChannel(event.getJDA(), userGuildId);
+        if (modMailAuditLog.isEmpty()) {
+            logger.warn(
+                    "Cannot find the designated modmail channel in server by id {} with the pattern modmail",
+                    userGuildId);
+            return;
+        }
+
+        MessageCreateAction message = createMessage(event, userId, modMailAuditLog.orElseThrow());
+
+        sendMessage(event, message);
+
+    }
+
+    private boolean handleIsOnCooldown(long userId, SlashCommandInteractionEvent event) {
+        if (isChannelOnCooldown(userId)) {
+            event.reply("Can only be used once per %s minutes.".formatted(COOLDOWN_DURATION_VALUE))
+                .setEphemeral(true)
+                .queue();
+            return true;
+        }
+        return false;
+    }
+
+    private Optional<TextChannel> getModMailChannel(JDA jda, long userGuildI) {
+        return jda.getGuildById(userGuildI)
+            .getTextChannelCache()
+            .stream()
+            .filter(channel -> modMailChannelNamePredicate.test(channel.getName()))
+            .findAny();
+    }
+
+    private MessageCreateAction createMessage(SlashCommandInteractionEvent event, long userId,
```

Comment:
which message? its a bit unclear since there are multiple messages involved:
* the message input from the user
* the message send to the mods
* the response shown to the user

maybe rename to `createModMessage`

Reply:
Done